tasks/eth/006-MCP-L1/postCheck.s.sol

```diff
@@ -0,0 +1,231 @@
+        postDeployAssertions();
+    }
+
+    /// @notice Asserts the correctness of an L1 deployment. This function expects that all contracts
+    ///         within the `prox` ContractSet are proxies that have been setup and initialized.
+    function postDeployAssertions() internal view {
+        console.log("Running post-deploy assertions");
+        ResourceMetering.ResourceConfig memory rcfg;
+        rcfg.maxResourceLimit = SystemConfig(prox.SystemConfig).resourceConfig().maxResourceLimit;
+        rcfg.elasticityMultiplier = SystemConfig(prox.SystemConfig).resourceConfig().elasticityMultiplier;
+        rcfg.baseFeeMaxChangeDenominator = SystemConfig(prox.SystemConfig).resourceConfig().baseFeeMaxChangeDenominator;
+        rcfg.minimumBaseFee = SystemConfig(prox.SystemConfig).resourceConfig().minimumBaseFee;
+        rcfg.systemTxMaxGas = SystemConfig(prox.SystemConfig).resourceConfig().systemTxMaxGas;
+        rcfg.maximumBaseFee = SystemConfig(prox.SystemConfig).resourceConfig().maximumBaseFee;
+        ResourceMetering.ResourceConfig memory dflt;
+        dflt.maxResourceLimit = Constants.DEFAULT_RESOURCE_CONFIG().maxResourceLimit;
+        dflt.elasticityMultiplier = Constants.DEFAULT_RESOURCE_CONFIG().elasticityMultiplier;
+        dflt.baseFeeMaxChangeDenominator = Constants.DEFAULT_RESOURCE_CONFIG().baseFeeMaxChangeDenominator;
+        dflt.minimumBaseFee = Constants.DEFAULT_RESOURCE_CONFIG().minimumBaseFee;
+        dflt.systemTxMaxGas = Constants.DEFAULT_RESOURCE_CONFIG().systemTxMaxGas;
+        dflt.maximumBaseFee = Constants.DEFAULT_RESOURCE_CONFIG().maximumBaseFee;
+        require(keccak256(abi.encode(rcfg)) == keccak256(abi.encode(dflt)));
+
+        checkSystemConfig();
+        checkL1CrossDomainMessenger();
+        checkL1StandardBridge();
+        checkL2OutputOracle();
+        checkOptimismMintableERC20Factory();
+        checkL1ERC721Bridge();
+        checkOptimismPortal();
+        checkProtocolVersions();
+        checkSuperchainConfig(false);
+    }
+
+    /// @notice Asserts that the SystemConfig is setup correctly
+    function checkSystemConfig() internal view {
+        console.log("Running chain assertions on the SystemConfig");
+        SystemConfig config = SystemConfig(prox.SystemConfig);
+
+        ResourceMetering.ResourceConfig memory resourceConfig;
+        resourceConfig.maxResourceLimit = config.resourceConfig().maxResourceLimit;
+        resourceConfig.elasticityMultiplier = config.resourceConfig().elasticityMultiplier;
+        resourceConfig.baseFeeMaxChangeDenominator = config.resourceConfig().baseFeeMaxChangeDenominator;
+        resourceConfig.minimumBaseFee = config.resourceConfig().minimumBaseFee;
+        resourceConfig.systemTxMaxGas = config.resourceConfig().systemTxMaxGas;
+        resourceConfig.maximumBaseFee = config.resourceConfig().maximumBaseFee;
+
+        require(config.owner() == cfg.finalSystemOwner());
+        require(config.overhead() == cfg.gasPriceOracleOverhead());
```

Comment:
Let's add reason strings to all `require` statements in this file, otherwise when one fails it will be hard to track down which failed. For brevity, I sometimes use incrementing numbers of 100 (count by 100 so you can add new ones in between without adjusting all numbers), for example:
```solidity
require(config.owner() == cfg.finalSystemOwner(), "100");
require(config.overhead() == cfg.gasPriceOracleOverhead(), "200");
```

then in the future, minimal diff to add a new check
```solidity
require(config.owner() == cfg.finalSystemOwner(), "100");
require(myNewCheck == cfg.myNewCheck(), "150");
require(config.overhead() == cfg.gasPriceOracleOverhead(), "200");
```